tests/test_modules/test_word_analysis/test_top_words.py

```diff
@@ -55,34 +55,25 @@ def test_top_words_with_saliency(simple_text_config, apply_mocked_startup_task,
     assert len(json_output.right) > 0 or len(json_output.errors) > 0
 
 
-def fake_no_saliency(indices):
-    records = []
-    for i in indices:
-        records.append(
-            TokensToWordsResponse(
-                saliency=16 * [0],  # 16 words
-                words=ALL_WORDS,
-            )
-        )
-    return records
-
-
-def test_top_words_without_saliency(monkeypatch, file_text_config_top1):
+def test_top_words_without_saliency(file_text_config_top1):
     save_predictions(file_text_config_top1)
 
     mod = TopWordsModule(
         dataset_split_name=DatasetSplitName.eval,
         config=file_text_config_top1,
         mod_options=ModuleOptions(top_x=16, pipeline_index=0),
     )
-
-    monkeypatch.setattr(mod, "get_words_saliencies", fake_no_saliency)
-    assert mod is not None
     [json_output] = mod.compute_on_dataset_split()
 
     top_word_all = [top_words_result.word for top_words_result in json_output.all]
+    top_word_right = [top_words_result.word for top_words_result in json_output.right]
+    top_word_errors = [top_words_result.word for top_words_result in json_output.errors]
 
-    for word in ["fred", "name", "sky", "blue"]:
-        assert word in top_word_all
-    for word in ["i", "is", "the", "."]:
-        assert word not in top_word_all
+    assert all(
+        word in top_word_all for word in ["phone", "potato", "pizza", "forest", "park", "turning"]
+    )
+    assert all(word not in top_word_all for word in ["to", "the", "not", "with", "on"])
+    assert all(word in top_word_right for word in ["phone", "potato", "pizza", "forest"])
+    assert all(word not in top_word_right for word in ["turning", "park"])
+    assert all(word in top_word_errors for word in ["phone", "turning", "park"])
+    assert all(word not in top_word_errors for word in ["potato", "pizza", "forest"])
```

Comment:
Here, you could leverage some sets to make the assertions more readable.
```suggestion
    assert {"phone", "turning", "park"}.issubset(top_word_errors)
    assert {"potato", "pizza", "forest"}.isdisjoint(top_word_errors)
```

If you change the comprehension lists to sets like so
```py
    top_word_all = {top_words_result.word for top_words_result in json_output.all}
    top_word_right = {top_words_result.word for top_words_result in json_output.right}
    top_word_errors = {top_words_result.word for top_words_result in json_output.errors}
```
you could even use `<=` (instead of `issubset()`) or `==` (if it applies here).
```suggestion
    assert {"phone", "turning", "park"} <= top_word_errors
    assert {"potato", "pizza", "forest"}.isdisjoint(top_word_errors)
```

Reply:
Ok cool! `sets` are still a bit of a mystery to me.